### ISSUE TITLE
[$500] IOU - IOU report is duplicated in LHN after emptying Merchant field and saving it

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#26205]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to staging.new.expensify.com
2. Request money from another use
3. Click on IOU preview two times to enter detail page
4. Click on the merchant
5. Empty the merchant and click Save

## Expected Result:
IOU report is not duplicated in LHN

## Actual Result:
IOU report is now duplicated in LHN. The duplicate disappears after the page is refreshed

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.58-1

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/5a1d9189-a03a-4972-88f3-f75a7e4fd739

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a14404ef51543e9e</li>
        <li>Upwork Job ID: 1697289310359842816</li>
        <li>Last Price Increase: 2023-09-07</li>
    </ul>
</details>

**Comment by StevenKKC** [comment]
## Proposal

### Please state again the problem we are trying to solve in this issue.
IOU report is duplicated in LHN after emptying Merchant field and saving it.

### What is the cause of this issue?
After emptying merchant field and saving it, BE returns error as below.
![Screenshot 2023-08-29 151043](https://github.com/Expensify/App/assets/131836113/2da2bbad-0b7c-466a-aae8-c0e8a054bd49)

So the following `failureData` is merged.
https://github.com/Expensify/App/blob/766dd656af85ae73057f4db02664d9b4b919ceae/src/libs/actions/IOU.js#L1020-L1038
But `iouReport.report` is `undefined` (this should be a typo), so `report_undefined` will be merged into `Onyx`.
Because `report_undefined`'s content is just same with `iouReport`, so this will be displayed in LHN.
Therefore, IOU report is duplicated.

### What changes do you think we should make to solve the problem?
We should change `iouReport.report` to `iouReport.reportID` in [this line](https://github.com/Expensify/App/blob/766dd656af85ae73057f4db02664d9b4b919ceae/src/libs/actions/IOU.js#L1035).

Merchant field should not be empty.
Thus we should add validating whether merchant field is empty or not in `EditRequestMerchantPage`.
* Add a validate function as below.
```js
    const validate = () => {
        const errorFields = {};
        if (_.isEmpty(merchantInputRef.current.value)) {
            errorFields.merchant = 'iou.error.emptyMerchant';
        }
        return errorFields;
    };
```
* Add `validate` `props` to `Form`.
```js
    <Form
        ...
        validate={validate}
```
* Add 'iou.error.emptyMerchant' translation key.
### What alternative solutions have you investigated? (Optional)
None.

**Comment by garrettmknight** [comment]
Might be related to https://github.com/Expensify/App/issues/24608#issuecomment-1688742711

**Comment by garrettmknight** [comment]
Cool, reproduced and I'll open it up!

**Comment by Pujan92** [comment]
I think will be fixed with this [PR](https://github.com/Expensify/App/pull/26373) as not allowing empty merchant.

**Comment by StevenKKC** [comment]
This https://github.com/Expensify/App/pull/26373 disallow empty merchant.
But if `EditMoneyRequest` API fails, still appear duplicated IOU report.
I think we should fix this bug.

**Comment by garrettmknight** [comment]
@jjcoffee What do you think?

**Comment by jjcoffee** [comment]
Hmm so looks like @Pujan92 is correct, that [PR](https://github.com/Expensify/App/pull/26373) _did_ fix the issue (it has been merged now and this particular issue is no longer reproducible), but @StevenKKC is also [correct](https://github.com/Expensify/App/issues/26205#issuecomment-1698031149) that there's a typo in the `failureData`, i.e. `iouReport.report` is used instead of `iouReport.reportID`, so if there is ever any error there will always be a duplicate report added as we'll be looking for an undefined report ID to merge into. I'm not sure how we'd normally handle this, should it be a new issue? @garrettmknight

**Comment by garrettmknight** [comment]
@jjcoffee Let's fix that issue here! @StevenKKC can you propose the fix?

**Comment by StevenKKC** [comment]
Updated [proposal](https://github.com/Expensify/App/issues/26205#issuecomment-1698031149).

**Comment by jjcoffee** [comment]
@StevenKKC's [proposed](https://github.com/Expensify/App/issues/26205#issuecomment-1698031149) fix LGTM!

:ribbon::eyes::ribbon: C+ reviewed

**Comment by garrettmknight** [comment]
@StevenKKC looks like the automation didn't work, but I've assigned you. Please get the PR up when you get a chance.

**Comment by StevenKKC** [comment]
@garrettmknight @jjcoffee PR https://github.com/Expensify/App/pull/27194 is ready for review.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by StevenKKC** [comment]
@garrettmknight PR was deployed to production, but the title was not updated and I haven't received an offer. Could you please take a look at this?

**Comment by garrettmknight** [comment]
@StevenKKC / @jjcoffee sorry for the wait on this one, slipped through the cracks with the 'Reviewing' label on it. Since this was such a small change, I'm going to forgo the urgency bonus.

Summarizing payments for this issue:
- Contributor: @StevenKKC $500
- Reviewer: @jjcoffee $500 paid via Upwork

Upwork job: https://www.upwork.com/en-gb/ab/applicants/1697289310359842816/job-details

**Comment by garrettmknight** [comment]
@StevenKKC Going to move this to monthly while your Upwork account gets sorted out. Ping me when it does and I'll complete payment.

**Comment by StevenKKC** [comment]
@garrettmknight I am afraid to let you know that My Upwork account is suspended so I couldn't get paid with my account. May I send proposal with my friend's account?

**Comment by garrettmknight** [comment]
Hey @StevenKKC we're only able to pay through your account since it's verified by Upwork. Once you've got your Upwork account working again I'll pay.
